### PR TITLE
Prague Region Competition Rules - new start list draw modes

### PR DESCRIPTION
**Pull request by lpechacek**
Regional championship rules impose certain principles on start list draw. http://psob.dig.cz/dokumenty/sr2013.pdf

Supersedes pull request #116.

### CODE REVIEW

**Comment by lpechacek** [comment]
I can see part of Martin's original contribution appearing in the master branch without proper attribution.  IMNSHO, users' contributions should be steered towards acceptable quality and not disassembled and reimplemented piece by piece without mentioning their name.

**Comment by fvacek** [comment]
Which part do you exactly mean and where?

**Comment by lpechacek** [comment]
[For the record, I agreed with Martin that I will adapt his patches for inclusion into QE.]

Yes,  dc3c283 (QE: extra_conditions in drawing helper functions) is my concern. The API change makes no sense by itself as there are no users of the functionality.

The change in question was originally part of Martin's patch. If you wanted to see the change split off into a separate patch, you should have said so. Reimplementing it in the way you did would also be acceptable if the the idea was attributed to Martin or there was a mention that the change is done in preparation of accepting his patch.

What I see gives me the impression of original contribution erosion with parts being pushed into the repository under the maintainer's name. Of course, it also makes my life harder as I have to rebase the patch set and remove the parts you already implemented.

**Comment by fvacek** [comment]
Please, look at my review notes in https://github.com/fvacek/quickbox/pull/116 and compare it with dc3c283 . Why should I make API changes without its utilization? Again, I just wanted to prepare better ground for Martin/You to have patch/merge process easier. Yes, I've implemented API change needed for merge request myself, because I'm QE designer, my change is simpler and shorter than proposed one and I'm short in time most of the night. If you feel that I've made review process violently shorter and better way is to propose my changes to you and find common consensus about that, than yes, it would be better if I would have time enough. That's all. If you have different impression, your impression is wrong.

Sorry guys, I have more important things to do than crying for 6 lines of code. If I've made detriment to anybody, sorry for that, I didn't want to.

**Comment by lpechacek** [comment]
I'm familiar with merge request #116 as I cleaned up the code. Your review comments were reflected in patch titled Minor refactoring which I merged into the one implementing the drawing options. And after all I like your patch with default parameter values by far the most elegant solution.

I was objecting the process of merging the change which makes Martin's contribution visually smaller and my job tougher as it's pushing me into rebasing, or merging origin/master, and solving the conflicts. Nothing more and nothing less.

Thanks for your efforts on QE development and openness to code contributions!

BTW the build check failed due to timeout. On my system the branch builds fine and start list drawing works as expected.

File: quickevent/app/plugins/qml/Runs/src/runswidget.h [hunk 1]
```diff
@@ -62,13 +62,18 @@ private slots:
 	QList< QList<int> > runnersByClubSortedByCount(int stage_id, int class_id, QMap<int, QString> &runner_id_to_club);
 	QList<int> runsForClass(int stage_id, int class_id);
```
Comment (lpechacek) [inline quickevent/app/plugins/qml/Runs/src/runswidget.h]: It looks to me like this comment refers to an outdated commit. With introduction of the API enhancement, code changes in runswidget.h are limited to new values in DrawMethod enum. The patch set no longer touches function prototypes. CMIIW

**Pull request merged**